urllib3/poolmanager.py

```diff
@@ -129,22 +129,32 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # Return False to re-raise any potential exceptions
         return False
 
-    def _new_pool(self, scheme, host, port):
+    def _new_pool(self, scheme, host, port, request_context=None):
         """
-        Create a new :class:`ConnectionPool` based on host, port and scheme.
+        Create a new :class:`ConnectionPool` based on host, port, scheme, and
+        any additional pool keyword arguments.
 
-        This method is used to actually create the connection pools handed out
-        by :meth:`connection_from_url` and companion methods. It is intended
-        to be overridden for customization.
+        If `request_context` is provided, it is provided as keyword arguments
+        to the pool class used. This method is used to actually create the
+        connection pools handed out by :meth:`connection_from_url` and
+        companion methods. It is intended to be overridden for customization.
         """
         pool_cls = self.pool_classes_by_scheme[scheme]
-        kwargs = self.connection_pool_kw
+        if request_context is None:
+            request_context = self.connection_pool_kw.copy()
+
+        # Although the context has everything necessary to create the pool,
+        # this function has historically only used the scheme, host, and port
+        # in the positional args. When an API change is acceptable these can
+        # be removed.
+        for key in ('scheme', 'host', 'port'):
+            request_context.pop(key, None)
```

Comment:
I didn't want to pass the scheme, host, or port from the context to the pool class since this function is still accepting those arguments as positional and the pool classes accept the host key as position and everything else as keyword arguments. They were [added to the context](https://github.com/shazow/urllib3/blob/master/urllib3/poolmanager.py#L170) since the pool key uses them. It is awkward to do here, though.